db/user.go

```diff
@@ -221,3 +222,71 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// ResolveUsers is a helper function to maps string representations of roles and * (all roles) to the emails for users with those roles.
+// The function takes an optional course and a list of strings, containing emails, roles, and * as input and returns an alphanumerically sorted slice of lowercase emails without duplicates.
+// A fatal error occurs if a course cannot be opened and the input contains roles or a *.
+// A warning will be logged for every invalid role.
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    if (backend == nil) {
+        return nil, fmt.Errorf("Database has not been opened.");
+    }
+
+    emailSet := map[string]any{};
+    roleSet := map[string]any{};
+
+    // Iterate over every given string, checking for emails, roles, and *, which denotes all users.
+    for _, email := range emails {
+        email = strings.ToLower(strings.TrimSpace(email));
+        if (email == "") {
+            continue;
+        }
+
+        if (strings.Contains(email, "@")) {
+            emailSet[email] = nil;
+        } else {
+            if (email == "*") {
+                allRoles := model.GetAllRoleStrings();
+                for role := range allRoles {
+                    roleSet[role] = nil;
+                }
+            } else {
+                roleSet[email] = nil;
+            }
+        }
+    }
+
+    if (len(roleSet) > 0) {
+        users, err := GetUsers(course);
+        if (err != nil) {
+            return nil, err;
+        }
+
+        for roleString := range roleSet {
+            if (roleString == "unknown") {
```

Comment:
This conditional is unnecessary.
The following conditional handles this case.

Remember that `model.RoleUnknown` is the zero value,
and is there fore returned from `model.GetRole()` when the result actually is "unknown" and when it is a role that is not known (e.g. "clowns").

Note that the unknown role is strictly a zero value and no user is every allowed to have that role.
So, even referring to the role (from outside the system) is not allowed.